projects/plugins/jetpack/extensions/blocks/top-posts/top-posts.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Top Posts Block.
+ *
+ * @since $$next-version$$
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Top_Posts;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	Blocks::jetpack_register_block(
+		__DIR__,
+		array( 'render_callback' => __NAMESPACE__ . '\load_assets' )
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Top Posts block registration/dependency declaration.
+ *
+ * @param array $attributes Array containing the Top Posts block attributes.
+ *
+ * @return string
+ */
+function load_assets( $attributes ) {
+	Jetpack_Gutenberg::load_assets_as_required( __DIR__ );
+
+	/*
+	 * We cannot rely on obtaining posts from the block because
+	 * top posts might have changed since then. As such, we must
+	 * make another request to check for updated stats.
+	 */
+	$request_url = sprintf(
```

Comment:
The request here wouldn't work on WPCOM, as `wpcom/v2` would be loaded to `public-api.wordpress.com`, so the API wouldn't be first party anymore.

I'm wondering whether the remote request could be saved by calling a function that returns top posts, which could save us the HTTP request overhead, and hence improve the performance of the block. More importantly, it would work with WP.COM with no issues. To accomplish this, we probably need to refactor the following code to a helper or something be shared by the route and the block.

https://github.com/Automattic/jetpack/pull/34153/files#diff-b82973540f31245c9564a0c3e997adba550cc9387fa7249be99581128844b45eR106

Reply:
Yes, we can leave it for now 😄

Reply:
PR for this in #34479 :)